iceoryx_posh/test/moduletests/test_capro_service.cpp

```diff
@@ -28,166 +30,653 @@ using namespace iox::capro;
-    EXPECT_TRUE(csd4 != csd4Ne);
+    EXPECT_THAT(serviceDescription1.getSourceInterface(), Eq(Interfaces::INTERNAL));
 }
 
-TEST_F(ServiceDescription_test, operatorAssign)
+TEST_F(ServiceDescription_test, ServiceDescriptionStringCtorCreatesServiceDescriptionWithValuesPassedToTheCtor)
 {
-    ServiceDescription csdNew = csd1;
-    EXPECT_TRUE(csdNew == csd1);
+    testService = "1";
+    testInstance = "2";
+    testEvent = "3";
+    ServiceDescription::ClassHash testHash = {1U, 2U, 3U, 4U};
+
+    ServiceDescription serviceDescription1 = ServiceDescription(testService, testInstance, testEvent, testHash);
+
+    EXPECT_THAT(serviceDescription1.getServiceIDString(), StrEq("1"));
+    EXPECT_THAT(serviceDescription1.getInstanceIDString(), StrEq("2"));
+    EXPECT_THAT(serviceDescription1.getEventIDString(), StrEq("3"));
+    EXPECT_EQ(uint16_t(1), serviceDescription1.getServiceID());
+    EXPECT_EQ(uint16_t(2), serviceDescription1.getInstanceID());
+    EXPECT_EQ(uint16_t(3), serviceDescription1.getEventID());
+    EXPECT_EQ(uint32_t(1), serviceDescription1.getClassHash()[0]);
+    EXPECT_EQ(uint32_t(2), serviceDescription1.getClassHash()[1]);
+    EXPECT_EQ(uint32_t(3), serviceDescription1.getClassHash()[2]);
+    EXPECT_EQ(uint32_t(4), serviceDescription1.getClassHash()[3]);
 }
+TEST_F(ServiceDescription_test, ServiceDescriptionStringCtorWithNonIntegerStringValuesSetTheIDsToInvalid)
+{
+    testService = "Service";
+    testInstance = "Instance";
+    testEvent = "Event";
+    ServiceDescription::ClassHash testHash = {1U, 2U, 3U, 4U};
+
+    ServiceDescription serviceDescription1 = ServiceDescription(testService, testInstance, testEvent, testHash);
 
-TEST_F(ServiceDescription_test, CreateServiceOnlyDescription)
+    EXPECT_EQ(InvalidID, serviceDescription1.getServiceID());
+    EXPECT_EQ(InvalidID, serviceDescription1.getInstanceID());
+    EXPECT_EQ(InvalidID, serviceDescription1.getEventID());
+}
+
+TEST_F(ServiceDescription_test, ServiceDescriptionStringCtorWithZeroAsStringValuesSetTheIDsToInvalid)
+{
+    testService = "0";
+    testInstance = "0";
+    testEvent = "0";
+    ServiceDescription::ClassHash testHash = {1U, 2U, 3U, 4U};
+
+    ServiceDescription serviceDescription1 = ServiceDescription(testService, testInstance, testEvent, testHash);
+
+    EXPECT_EQ(InvalidID, serviceDescription1.getServiceID());
+    EXPECT_EQ(InvalidID, serviceDescription1.getInstanceID());
+    EXPECT_EQ(InvalidID, serviceDescription1.getEventID());
+}
+
+TEST_F(ServiceDescription_test, ServiceDescriptionStringCtorWithOutOfBoundaryIntegerStringValuesSetTheIDsToInvalid)
 {
-    ServiceDescription desc1(1u, 2u);
-    ServiceDescription desc2("bla", "fuh");
-    ServiceDescription desc3(1u, 2u, 3u);
-    ServiceDescription desc4("bla", "fuh", "dumbledoodoo");
+    /// @note The value 65536 is assigned to the strings to set the condition to out of boundary when string is
+    /// converted to uint_16t.
+    testService = "65536";
```

Comment:
Instead of a comment try to write more intentional code. for instance to make it clear that you would like to have a uint16_t overflow you could write
```cpp
testService = std::to_string(uint32_t(1) + std::numeric_limits<uint16_t>::max() );
```

Reply:
I'm getting error if I'm trying to do this way.

no match for ‘operator=’ (operand types are ‘iox::capro::IdString_t {aka iox::cxx::string<100>}’ and ‘std::__cxx11::string {aka std::__cxx11::basic_string<char>}

Reply:
You need to create a `cxx::string` from the `std::string`, here is described how to do this: https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_utils/include/iceoryx_utils/cxx/string.hpp#L204
You can directly inject the std::to_string statement in the constructing call for the `cxx::string`

Reply:
yeah, I just found it out :)